src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyNode<T> first;
+    private MyNode<T> last;
+
+    public class MyNode<T> {
+        T item;
+        MyNode<T> next;
+        MyNode<T> prev;
+
+        public MyNode(MyNode<T> prev, T item, MyNode<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (size == 0) {
+            last = new MyNode<T>(null, value, null);
+            first = last;
+        } else {
+            MyNode<T> addedElement = new MyNode<T>(last, value, null);
+            last.next = addedElement;
+            last = addedElement;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
```

Comment:
let's move this check to a separate method, we use it in different places in our code